tests/test_x2pdf.py

```diff
@@ -9,23 +9,42 @@
 from .conftest import run_cli
 
 
-def test_x2pdf(capsys, tmp_path: Path) -> None:
+def test_x2pdf_with_format(capsys, tmp_path: Path) -> None:
     # Arrange
     output = tmp_path / "out.pdf"
     assert not output.exists()
+    
+    formats_to_test = [
+        "Letter",
+        "A4-portrait",
+        "A4-landscape",
+        "210x297",
+        "invalid-format"
+    ]
+    
+    for format_option in formats_to_test:
+        # Act
+        exit_code = run_cli(
+            [
+                "x2pdf",
+                "sample-files/003-pdflatex-image/page-0-Im1.jpg",
```

Comment:
The tricky part is when there are several images with variying size in the document:
could you add tests for those cases please?

Also, it would be interesting to check how `pdfly x2pdf` behaves when asked to use `--format A4` and an image in landscape orientation is provided (it be centered on the page)